### PR TITLE
Updated tool spot_detection_2d: LoG filter included

### PR DESCRIPTION
**Pull request by qgao-hd**

### CODE REVIEW

File: tools/spot_detection_2d/spot_detection_2d.xml [hunk 1]
```diff
@@ -1,35 +1,40 @@
...
     </command>
     <inputs>
         <param name="fn_in" type="data" format="tiff" label="Image sequence (stack)" />
         <param name="frame_1st" type="integer" value="1" label="Starting time point (1 for the first frame of the stack)" />
         <param name="frame_end" type="integer" value="0" label="Ending time point (0 for the last frame of the stack)" />
+        <param name="filter" type="select" label="Choose a detection filter">
+            <option value="Gauss" selected="True">Gaussian</option>
+            <option value="LoG">Laplacian-of-Gaussian, LoG (more accurate when spots have similar sizes)</option>
+        </param>
+        <param name="ssig" type="float" value="1" label="Sigma of the Gaussian (for noise suppression)" />
```
Comment (bgruening) [inline tools/spot_detection_2d/spot_detection_2d.xml:31]: 1.0? adding min and max maybe?

File: tools/spot_detection_2d/spot_detection_2d.xml [hunk 2]
```diff
@@ -1,35 +1,40 @@
...
+        <param name="thres" type="float" value="10" label="Percentage of the global maximal as the threshold for candidate spots" />
```
Comment (bgruening) [inline tools/spot_detection_2d/spot_detection_2d.xml:32]: value="10.0" min="0" max="100"?

File: tools/spot_detection_2d/spot_detection_2d.xml [hunk 3]
```diff
@@ -39,16 +44,19 @@
...
             <param name="frame_end" value="0"/>
-            <param name="typ_intens" value="smoothed"/>
-            <param name="thres" value="10"/>
+            <param name="filter" value="Gauss"/>
             <param name="ssig" value="1"/>
+            <param name="thres" value="10"/>
+            <param name="typ_intens" value="smoothed"/>
             <param name="bndy" value="10"/>
-            <output name="fn_out" value="spots_detected.tsv" ftype="tabular" />
+            <output name="fn_out" value="spots_detected.tsv" ftype="tabular" compare="diff" />
```
Comment (bgruening) [inline tools/spot_detection_2d/spot_detection_2d.xml:52]: diff is default method, so you do not need to specify it

File: tools/spot_detection_2d/spot_detection_2d.xml (hunk 1 above)
Comment (qgao-hd) [reply tools/spot_detection_2d/spot_detection_2d.xml]: Updated. Thanks

File: tools/spot_detection_2d/spot_detection_2d.xml (hunk 2 above)
Comment (qgao-hd) [reply tools/spot_detection_2d/spot_detection_2d.xml]: Updated. Thanks

**Pull request merged**